src/kdl_parser.cpp

```diff
@@ -171,5 +170,247 @@ bool treeFromUrdfModel(const urdf::ModelInterface& robot_model, Tree& tree)
     return true;
 }
 
+/*
+ * SDF to KDL
+ * bellow there are the source codes to convert SDFs information to KDL
+ */
+
+typedef std::map<std::string, sdf::ElementPtr> JointsMap;
+typedef std::map<std::string, sdf::ElementPtr> LinksMap;
+typedef std::map<std::string, std::vector<std::string> > LinkNamesMap;
+
+/**
+ * convert <axis> element to KDL::Vector
+ */
+KDL::Vector sdf_axis_to_kdl(sdf::ElementPtr sdf)
+{
+    double x, y, z;
+    std::string axis_value = sdf->GetElement("xyz")->Get<std::string>();
+    sscanf(axis_value.c_str(), "%lf %lf %lf", &x, &y, &z);
+    return KDL::Vector(x, y, z);
+}
+
+/**
+ * convert <pose> element to KDL::Frame
+ */
+KDL::Frame sdf_pose_to_kdl(sdf::ElementPtr sdf)
+{
+
+    double x, y, z;
+    double roll, pitch, yaw;
+    std::string pose_value = sdf->Get<std::string>();
+
+    sscanf(pose_value.c_str(), "%lf %lf %lf %lf %lf %lf", &x, &y, &z, &roll, &pitch, &yaw);
+
+    KDL::Rotation rotation = KDL::Rotation::RPY(roll, pitch, yaw);
+    KDL::Vector position = KDL::Vector(x, y, z);
+
+    return KDL::Frame(rotation, position);
+}
+
+/**
+ * convert <inertial> element to KDL::RigidBodyInertia
+ */
+KDL::RigidBodyInertia sdf_inertial_to_kdl(sdf::ElementPtr sdf)
+{
+
+    KDL::Frame pose = sdf_pose_to_kdl(sdf->GetElement("pose"));
+    double mass;
+
+    if (sdf->HasElement("mass")){
+        mass = sdf->GetElement("mass")->Get<double>();
+        if (sdf->HasElement("inertia")){
+
+            sdf::ElementPtr sdf_inertia = sdf->GetElement("inertia");
+
+            char *tags[] = {"ixx", "ixy", "ixz", "iyy", "iyz", "izz" };
+
+            double ixx, ixy, ixz, iyy, iyz, izz;
+            double *ptr[] = {&ixx, &ixy, &ixz, &iyy, &iyz, &izz};
+
+            for (int i = 0; i < 6; i++){
+                (*ptr[i]) = sdf_inertia->GetElement(tags[i])->Get<double>();
+            }
+
+            return pose.M * RigidBodyInertia(mass, pose.p,
+                                            RotationalInertia(ixx, iyy, izz, ixy, ixz, iyz));
+        }
+    }
+
+    return KDL::RigidBodyInertia();
+}
+
+/**
+ * convert <joint> element to KDL::Joint
+ */
+KDL::Joint sdf_joint_to_kdl(sdf::ElementPtr sdf){
```

Comment:
wrong naming convention ... you (and the rest of the code) uses lowerCamelCase everywhere else.